### PR TITLE
Greg/668/single signature

### PR DESCRIPTION
**Pull request by ghubertpalo**
## Content
This PR includes adds a `SingleVerificationKey` type.

## Pre-submit checklist

- Branch
  - [x] Tests are provided (if possible)
  - [x] Crates versions are updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested
- Documentation
  - [ ] Update README file (if relevant)
  - [ ] Update documentation website (if relevant)
  - [ ] Add dev blog post (if relevant)

## Comments
More types may be refactored in this ticket.

## Issue(s)
Relates to #668

### CODE REVIEW

File: mithril-aggregator/src/database/provider/single_signature.rs [hunk 1]
```diff
@@ -38,30 +38,39 @@ pub struct SingleSignatureRecord {
...
+        let signatures = SingleSignatures {
+            party_id: value.signer_id,
+            won_indexes: value.lottery_indexes,
+            signature: value.signature.try_into()?,
+        };
+
+        Ok(signatures)
     }
+
+    // fn try_from(other: SingleSignatureRecord) -> SingleSignatures {
```
Comment (Alenar) [inline mithril-aggregator/src/database/provider/single_signature.rs:72]: Remove this dead code ;)

File: mithril-aggregator/src/http_server/routes/signatures_routes.rs [hunk 2]
```diff
@@ -58,7 +58,15 @@ mod handlers {
 
         match signed_entity_type {
             Ok(signed_entity_type) => {
-                let signature = FromRegisterSingleSignatureAdapter::adapt(message);
+                let signature = match FromRegisterSingleSignatureAdapter::try_adapt(message) {
+                    Err(err) => {
+                        warn!("Could not parse single signature message.");
+
+                        return Ok(reply::internal_server_error(err.to_string()));
```
Comment (Alenar) [inline mithril-aggregator/src/http_server/routes/signatures_routes.rs:65]: Use a bad request and an error message aligned with what we use below, ie:

```rust
let signer = match FromRegisterSingleSignatureAdapter::try_adapt(message) {
    Ok(signature) => signature,
    Err(err) => {
        warn!("register_signatures::payload decoding error"; "error" => ?err);
        return Ok(reply::bad_request(
            "Could not decode signature payload".to_string(),
            err.to_string(),
        ));
    }
};
```

File: mithril-signer/src/message_adapters/to_register_signature_message.rs [hunk 3]
```diff
@@ -12,7 +12,7 @@ impl ToMessageAdapter<(SignedEntityType, SingleSignatures), RegisterSignatureMes
         RegisterSignatureMessage {
             signed_entity_type: Some(signed_entity_type),
             party_id: single_signature.party_id,
-            signature: single_signature.signature,
+            signature: single_signature.signature.try_into().unwrap(),
```
Comment (jpraynaud) [inline mithril-signer/src/message_adapters/to_register_signature_message.rs:15]: Idea: implement a `try_adapt` to avoid signer node panic

**Review by jpraynaud — APPROVED** [review]
LGTM :+1:

**Pull request merged**